docs/FeatureManagement/FeatureFlag.v2.0.0.schema.json

```diff
@@ -0,0 +1,345 @@
+                "required": [],
+                "patternProperties": {
+                  "^.*$": {
+                    "anyOf": [
+                      {
+                        "type": "string"
+                      },
+                      {
+                        "type": "null"
+                      },
+                      {
+                        "type": "object"
+                      },
+                      {
+                        "type": "number"
+                      },
+                      {
+                        "type": "array"
+                      },
+                      {
+                        "type": "boolean"
+                      }
+                    ]
+                  }
+                },
+                "additionalProperties": false
+              }
+            }
+          }
+        }
+      }
+    },
+    "variants": {
+      "$id": "#/properties/variants",
+      "type": "array",
+      "title": "Variants Collection",
+      "description": "The list of variants defined for this feature. A variant represents a value of a feature flag that can be a string, a number, a boolean, or even a configuration object.",
+      "items": {
+          "$id": "#/properties/variants/items",
+          "type": "object",
+          "title": "Variant",
+          "required": [
+              "name"
+          ],
+          "properties":{
+              "name": {
+                  "$id": "#/properties/variants/items/properties/name",
+                  "type": "string",
+                  "title": "Variant Name",
+                  "description": "The name used to refer to a feature variant.",
+                  "pattern": "^(.*)$"
+              },
+              "configuration_value": {
+                  "$id": "#/properties/variants/items/properties/configuration_value",
+                  "type": ["string", "null", "number", "object", "array", "boolean"],
+                  "title": "Variant Configuration Value",
+                  "description": "The inline value set for this feature variant.",
+                  "default": null
+              },
+              "configuration_reference": {
+                  "$id": "#/properties/variants/items/properties/configuration_reference",
+                  "type": "string",
+                  "title": "Variant Configuration Reference",
+                  "description": "The string used to reference a path in the configuration and set the variant's value equal to the value at that path.",
+                  "pattern": "^[a-zA-Z0-9]+(:[a-zA-Z0-9]+)*$"
+              },
+              "status_override": {
+                  "$id": "#/properties/variants/items/properties/status_override",
+                  "type": "string",
+                  "title": "Variant Status Override",
+                  "description": "Overrides the enabled state of the feature according to this value when this variant is used. Does not override the state if value is None.",
```

Comment:
How do you like this?

Overrides the enabled state of the feature if the given variant is assigned.

Reply:
Right, since we agreed to use the term "assigned" now I think this makes it clearer.